config/routes.rb

```diff
@@ -0,0 +1,15 @@
+Rails.application.routes.draw do
+  root "trips#index"
+
+  resources :trips, :drivers, :passengers
+
+  # Nested routes for drivers and passengers to have access to associated trips
+  resources :drivers do
+    resources :trips, only: [:index, :new]
```

Comment:
Do you ever show a trips index page in the context of a driver or passenger?  Do you ever create a new trip in the context of a driver?  Driver's don't normally initiate the rideshare process.

Reply:
Hmm I think i'm still a bit confused on nested routes. Would it have made more sense for these to be in reverse order where the nested routes are :drivers and :passengers, only [:index, :new]/trips being the outer resources? Would that mean that creating a new driver/passenger would be in the context of a trip/trips are initiating the rideshare process rather than vice versa?

Reply:
@cassyarchibald 
I don't think you ever need a `:new` action for trips, as you don't really need a form for it.  I'm not sure you even need an index page for trips, as we mostly just show trips in the driver and passenger show pages.